test/spec/app_spec.rb

```diff
@@ -433,25 +432,168 @@
-    #   skip("Needs to be implemented")
-    # end
+    it "should edit basic source data (name, description/bio, avatar)", bin2: true do
+      api_create_team_project_and_source_and_redirect_to_source('ACDC', 'https://twitter.com/acdc')
+      sleep 5
+      element = @driver.find_element(:css, '.source__tab-button-notes')
+      element.click
+      sleep 1 until element = @driver.find_element(:class, "source__edit-button")
+      element.click
+      input = @driver.find_element(:id, 'source__name-container')
+      input.send_keys(" - EDIT ACDC")
+      input = @driver.find_element(:id, 'source__bio-container')
+      input.send_keys(" - EDIT DESC")
+      @driver.find_element(:class, "source__edit-avatar-button").click
+      sleep 1
+      input = @driver.find_element(:css, 'input[type=file]')
+      input.send_keys(File.join(File.dirname(__FILE__), 'test.png'))
+      sleep 1
+      @driver.find_element(:class, 'source__edit-save-button').click
+      sleep 5 
+      displayed_name = get_element('h1.source__name').text
+      expect(displayed_name.include? "EDIT").to be(true)
+    end
 
-    # it "should add and remove accounts to sources" do
-    #   skip("Needs to be implemented")
-    # end
+    it "should add and remove accounts to sources", bin2: true  do
+      api_create_team_project_and_source_and_redirect_to_source('GOT', 'https://twitter.com/GameOfThrones')
+      sleep 5
+      element = @driver.find_element(:class, "source__edit-button")
+      element.click
+      sleep 3 
+      @driver.find_element(:class, "source__edit-addinfo-button").click
+      sleep 1
+      @driver.find_element(:class, "source__add-link").click
+      sleep 1
+      fill_field("source__link-input0", "www.acdc.com", :id)
+      sleep 2
+      @driver.find_element(:class, 'source__edit-save-button').click
+      sleep 5 
+      expect(@driver.page_source.include?('AC/DC Official Website')).to be(true)
+      #delete
+      element = @driver.find_element(:class, "source__edit-button")
+      element.click
+      sleep 3
+      list = @driver.find_elements(:css => "svg[class='create-task__remove-option-button create-task__md-icon']")
+      list[1].click
+      sleep 1
+      @driver.find_element(:class, 'source__edit-save-button').click
+      sleep 5 
+      expect(@driver.page_source.include?('AC/DC Official Website')).to be(false)
+    end
 
-    # it "should edit source metadata (contact, phone, location, organization, other)" do
-    #   skip("Needs to be implemented")
-    # end
+    it "should edit source metadata (contact, phone, location, organization, other)", bin2: true do    
+      api_create_team_project_and_source_and_redirect_to_source('GOT', 'https://twitter.com/GameOfThrones')
+      sleep 5
+      element = @driver.find_element(:class, "source__edit-button")
+      element.click
+      sleep 1 
+      @driver.find_element(:class, "source__edit-buttons-add-merge").click
```

Comment:
this class should preferably be `source__edit-addinfo-button` instead of `source__edit-buttons-add-merge`